grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1140,110 +1148,441 @@ class StockMovementService {
+                                    "Unable to find internal location with name {0}")
+                        }
+                    }
+
+                    // FIXME It's probably ok to use product availability data here. However, we
+                    //  probably want to validate against actual inventory items (calculated from
+                    //  transactions) before the stock movement is shipped (i.e. transactions created).
+                    //  Otherwise we could encounter a situation where the picklist item is valid based on
+                    //  the product availability, but this data might be stale for some reason
+
+                    // TODO Need to test whether this satisfies the requirement when passing a NULL
+                    //  bin location. We also probably want to deal with the case where the provided
+                    //  bin location is provided, but not found.
+
+                    // If there was a bin location provided then we'll want to check if there's an
+                    // available item at that location
+                    if (data.binLocation) {
+
+                        // Let's try to determine whether there's a specific available inventory item
+                        //  associated with the lot number and bin data provided by the user.
+                        AvailableItem availableItem = pickPageItem.getAvailableItem(inventoryItem, internalLocation)
+
+                        // FIXME This is only necessary if we cannot find a happy case from OBPIH-6331
+                        if (!availableItem) {
+                            ApplicationTagLib g = grailsApplication.mainContext.getBean(ApplicationTagLib.class)
+                            String lotNumberName = data.lotNumber ?: g.message(code: "default.noLotNumber.label", default: Constants.DEFAULT_LOT_NUMBER)
+                            String binLocationName = data.binLocation ?: g.message(code: "default.noBinLocation.label", default: Constants.DEFAULT_BIN_LOCATION_NAME)
+                            data.errors.rejectValue(
+                                    "id",
+                                    "importPickCommand.availableItem.notFound.error",
+                                    [lotNumberName, binLocationName] as Object[],
+                                    "There is no available item for lot number {0} and bin location {1}"
+                            )
+                        }
+
+                        // If there is an available item for the location but it's unavailable
+                        if (availableItem && !availableItem.isQuantityPickable(data?.quantity)) {
+                            String lotNumberName = data.lotNumber ?: g.message(code: "default.noLotNumber.label", default: Constants.DEFAULT_LOT_NUMBER)
+                            String binLocationName = data.binLocation ?: g.message(code: "default.noBinLocation.label", default: Constants.DEFAULT_BIN_LOCATION_NAME)
```

Comment:
This was probably an issue that came after rebase but it looks like `g` is not defined in this scope so we need to add 
```groovy
ApplicationTagLib g = grailsApplication.mainContext.getBean(ApplicationTagLib.class)
``` 
in the scope or move this same code from the previous `if (!availableItem) ` in a higher block/scope (move it out of the if block)

Reply:
yeah this was my fault. I think I moved it into the wrong block. Fixed!